packages/components/src/navigation/item.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Icon, chevronRight } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import Text from '../text';
+
+const Item = ( { data, item, setActive, isActive } ) => {
+	const children = data.filter( ( d ) => d.parent === item.slug );
+	const onSelect = () => {
+		const next = children.length ? children[ 0 ] : item;
+		setActive( next );
```

Comment:
>Part of me is thinking that the whole nav component should remove stateless

I feel this way too. I'd like to strive to have a "dumb" component as much as possible. I added state here because I was thinking of a scenario where there are multiple depths of menus which require the user to drill down and the page may not change until the user gets to the bottom. In this case the component would need to keep track of where it is in the tree. 

But now I think this may be an edge case we can ignore for now. One solution would be to require at least one "page" for every level. For instance, the "Home" link below:

![Screen Shot 2020-07-23 at 2 12 39 PM](https://user-images.githubusercontent.com/1922453/88246450-aa51ed00-ccee-11ea-8de2-e6f391faddb3.png)

I will remove state and simplify this component next.